ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1277,19 +1291,21 @@ private void prepareForSending() {
       try {
         BlobDataType blobDataType =
             isMetadataChunk() ? BlobDataType.METADATA : isSimpleBlob ? BlobDataType.SIMPLE : BlobDataType.DATACHUNK;
-        partitionId = getPartitionForPut(partitionClass, attemptedPartitionIds);
-        // To ensure previously attempted partitions are not retried for this PUT after a failure.
-        attemptedPartitionIds.add(partitionId);
 
         if (shouldUseReservedMetadataId()) {
+          partitionId = metadataPutChunk.reservedMetadataChunkId.getPartition();
```

Comment:
Ankur, the PR looks good to me.

Just wondering,
Before the fix, the chunkBlobId and partitionId are from two different partitions? What's the consequence or behavior after it?

Reply:
The consequence was that Puts were failing.
OperationTracker took the blobid of the reserved metadata, and in some cases a different partition id. And so the put request landed on hosts that didnt have the partition in the blobid. These put requests just failed.